### ISSUE TITLE
warewulf-rockylinux:8 container missing /run/systemd/system, leading to no telegraf.service

### ISSUE THREAD

**Issue by NateCrawford** [hpcng/warewulf#826]
### Version of Warewulf

What version of Warewulf are you using? 

`wwctl version:	 4.4.0-1.git_afcdb21
rpc version: apiPrefix:"rc1" apiVersion:"1" warewulfVersion:"4.4.0-1.git_afcdb21"
`

### Expected behavior

On entering ww-rocky-8 container, expect to have /run/systemd/system directory

### Actual behavior

The /run/systemd/system directory is missing. This causes some packages to believe systemd is not being used, and then skip installing their systemd service scripts.

### Steps to reproduce this behavior
wwctl container import docker://ghcr.io/hpcng/warewulf-rockylinux:8 test-r8
wwctl container shell test-r8
ls -l /run/systemd/system

### What OS/distro are you running

```
NAME="Rocky Linux"
VERSION="8.7 (Green Obsidian)"
ID="rocky"
ID_LIKE="rhel centos fedora"
VERSION_ID="8.7"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Rocky Linux 8.7 (Green Obsidian)"
ANSI_COLOR="0;32"
LOGO="fedora-logo-icon"
CPE_NAME="cpe:/o:rocky:rocky:8:GA"
HOME_URL="https://rockylinux.org/"
BUG_REPORT_URL="https://bugs.rockylinux.org/"
ROCKY_SUPPORT_PRODUCT="Rocky-Linux-8"
ROCKY_SUPPORT_PRODUCT_VERSION="8.7"
REDHAT_SUPPORT_PRODUCT="Rocky Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="8.7"

```

### How did you install Warewulf

Write here how you installed Warewulf. Eg. RPM, source.

Built RPMs from source.

### Additional notes:
The problem surfaced when trying to install telegraf-1.26.2-1.x86_64 from the Influxdata repo. The package installs without error, but does not install and enable /usr/lib/systemd/system/telegraf.service. 

Looking at (https://github.com/influxdata/telegraf/blob/master/scripts/rpm/post-install.sh), the existence of this directory seems important for proper detection of systemd being in control:
```
# Set up systemd service - check if the systemd directory exists per:
# https://www.freedesktop.org/software/systemd/man/sd_booted.html
if [[ -d /run/systemd/system ]]; then
    cp -f /usr/lib/telegraf/scripts/telegraf.service /usr/lib/systemd/system/telegraf.service
    systemctl enable telegraf
    systemctl daemon-reload
fi
```
Is there a proper way to fake systemd actually running in a container? Is `mkdir /run/systemd/system` enough?

**Comment by anderbubble** [comment]
@NateCrawford on first read, I think I consider this is a bug in telegraf's package. If I could, I'd rewrite their post-install like this:

```
if [[ -d /usr/lib/systemd/system ]]; then
    cp -f /usr/lib/telegraf/scripts/telegraf.service /usr/lib/systemd/system/telegraf.service
fi
if [[ -d /run/systemd/system ]]; then
    systemctl enable telegraf
    systemctl daemon-reload
fi
```

`/run/systemd/system` only exists when systemd is _running_, which it is when a cluster node is booted but not when a container is being built or wwctl has shelled into it. telegraf would have this same issue in any other container environment; e.g., if you installed telegraf in a Docker container, it _also_ wouldn't place the unit file or enable the service. Difference there is that you likely don't need it to.

If I were in your circumstance, I'd probably just capture the necessary post-install bits into the Containerfile for your container, rather than try to fool it into doing the right thing.

```
FROM docker://ghcr.io/hpcng/warewulf-rockylinux:8

COPY influxdb.repo /etc/yum.repos.d/

RUN dnf -y install telegraf \
     && cp -f /usr/lib/telegraf/scripts/telegraf.service /usr/lib/systemd/system/telegraf.service \
     && systemctl enable telegraf
```

Doing a `mkdir /run/systemd/system` might work here; but it will likely cause other side effects. For example, `systemctl daemon-reload` would fail in the post-install, which might cause more problems; and you might find other packages behave _worse_ when they are told that systemd is running but isn't.

**Comment by NateCrawford** [comment]
I tend to agree, and will submit an issue with the telegraf people. They implemented this test last Fall to try to do the systemd test in a distro-agnostic way, but only testing the "sd_booted" status was too simple a solution.

For reference, simply making the directory works in this case, but provokes many more warnings during the install. 

I'm used to rpm installations throwing out dbus warnings when installing inside a container, and it would be nice if systemd had a way to tell it it was in a container, but I'm not counting on that.

Thanks!

**Comment by NateCrawford** [comment]
Cool! 

Also, do NOT just 'mkdir /run/systemd/system', as this breaks 'systemctl enable/disable servicename'.

**Comment by NateCrawford** [comment]
Hmm, "systemd-detect-virt -r" returns 1 both inside and outside the container. "systemd-detect-virt" with no options returns "none" in both cases for me. This is with systemd-239-68.el8_7.4. I'll check if this is a known issue.

**Comment by NateCrawford** [comment]
Yep, known issue, fixed in 2021 (v248): https://github.com/systemd/systemd/pull/17902

**Comment by mslacken** [comment]
You could add `/run` to the mounts sections in `warewulf.conf` so that it gets mounted into the container from the outside host. The problem here could be that you interact as root with the systemd of the host.

**Comment by griznog** [comment]
> Yep, known issue, fixed in 2021 (v248): [systemd/systemd#17902](https://github.com/systemd/systemd/pull/17902)

Sorry, should have mentioned I've only used this on Rocky 8. Didn't realize it was so new-ish.
